src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -590,17 +590,30 @@ def container_export(cmd, resource_group_name, name, file):
                                              name,
                                              container_group_client.api_version,
                                              False).__dict__
-
     # Remove unwanted properites
     resource['properties'].pop('instanceView', None)
     resource.pop('sku', None)
     resource.pop('id', None)
     resource.pop('plan', None)
-    resource.pop('identity', None)
     resource.pop('kind', None)
     resource.pop('managed_by', None)
     resource['properties'].pop('provisioningState', None)
 
+    # Correctly export the identity
+    if 'identity' in resource and resource['identity'].type != ResourceIdentityType.none:
+        resource['identity'] = resource['identity'].__dict__
+        resource['identity'].pop('additional_properties', None)
+        resource['identity'].pop('principal_id', None)
+        resource['identity'].pop('tenant_id', None)
+
+        resource['identity']['type'] = resource['identity']['type'].value
+
+        for key in resource['identity']['user_assigned_identities'] or {}:
```

Comment:
What is the purpose of `or {}:` in this case? If `resource['identity']['user_assigned_identities']` was empty, this loop wouldn't run anyways (and still wouldn't if it hit `or {}`)?

```python
>>> for key in {} or {}:
...     print('here')
... 
>>>

```

Reply:
if 'user-assinged_identities' is empty it would return None not {}, In that case, python would through an exception saying that None is not iterable. The or would protect against that.

Reply:
@samkreter 

Ah, of course! Thanks!